notebooks/ch18/hyp_studentized.md

```diff
@@ -0,0 +1,415 @@
+import pandas as pd
+import seaborn as sns
+%matplotlib inline
+import ipywidgets as widgets
+from ipywidgets import interact, interactive, fixed, interact_manual
+import nbinteract as nbi
+
+sns.set()
+sns.set_context('talk')
+np.set_printoptions(threshold=20, precision=2, suppress=True)
+pd.options.display.max_rows = 7
+pd.options.display.max_columns = 8
+pd.set_option('precision', 2)
+# This option stops scientific notation for pandas
+# pd.set_option('display.float_format', '{:.2f}'.format)
+```
+
+
+```python
+# HIDDEN
+def df_interact(df, nrows=7, ncols=7):
+    '''
+    Outputs sliders that show rows and columns of df
+    '''
+    def peek(row=0, col=0):
+        return df.iloc[row:row + nrows, col:col + ncols]
+    if len(df.columns) <= ncols:
+        interact(peek, row=(0, len(df) - nrows, nrows), col=fixed(0))
+    else:
+        interact(peek,
+                 row=(0, len(df) - nrows, nrows),
+                 col=(0, len(df.columns) - ncols))
+    print('({} rows, {} columns) total'.format(df.shape[0], df.shape[1]))
+```
+
+
+```python
+# Generate a bunch of random points. To change the dataset, just
+# change this line and rerun notebook
+times = (np.random.exponential(scale=300, size=13825) +
+         np.abs(np.random.normal(loc=50, scale=40, size=13825)))
+```
+
+## The Studentized Bootstrap
+
+
+
+The **bootstrap** is a process we learned about in Data 8 that we can use for estimating a population statistic using only one sample. The general procedure for bootstrapping is as follows:
+- Draw a sizable sample from the population.
+- Using this sample, we *resample with replacement* to form new samples of the same size.
+- We do this many times, taking the desired statistic of each resample (e.g. we take the median of each resample).
+
+Here, we end up with many test statistics from individual resamples, from which we can form a distribution. In Data 8, we were taught to form a confidence interval by taking the 2.5th percentile and the 97.5th percentile. This method of bootstrapping to create a confidence interval is called the **percentile bootstrap.** Generally speaking, this confidence interval will have a 95% chance of containing the actual population statistic. It is important to note, however, that this will NOT always be the case. 
+
+Below, we've taken a population and created one thousand bootstrap *confidence intervals* for the population mean for different sample sizes. The y-axis represents the fraction of the one thousand confidence intervals that contained the real population mean. Notice that at sample sizes below 20, fewer than 90% of the confidence intervals actually contain the population mean.
+
+
+```python
+# Since this plot needs a bunch of prior code to run, I suggest
+# saving this plot as a .png and then just including the picture without the
+# code in the final notebook
+# trials['percentile'].plot()
+# plt.axhline(0.95, c='red', linestyle='--', label='95% coverage');
+```
+
+<img src="trials.png">
```

Comment:
Note that this image will break when published since the notebook will be converted to HTML and the HTML file will be moved somewhere else. Instead, you should directly upload the image to the assets folder: https://github.com/DS-100/textbook/tree/master/assets and use the raw URL of the image there.